src/simtools/production_configuration/limits_calculation.py

```diff
@@ -1,63 +1,110 @@
+
+    def _read_event_data(self):
+        """Read the event data from the HDF5 file."""
+        with h5py.File(self.event_data_file, "r") as f:
+            if "data" in f:
+                data_group = f["data"]
+                try:
+                    self.event_x_core = data_group["core_x"][:]
+                    self.event_y_core = data_group["core_y"][:]
+                    self.simulated = data_group["simulated"][:]
+                    self.shower_id_triggered = data_group["shower_id_triggered"][:]
+                    self.list_of_files = data_group["file_names"][:]
+                    self.shower_sim_azimuth = data_group["shower_sim_azimuth"][:]
+                    self.shower_sim_altitude = data_group["shower_sim_altitude"][:]
+                    self.array_altitude = data_group["array_altitude"][:]
+                    self.array_azimuth = data_group["array_azimuth"][:]
+                    self.trigger_telescope_list_list = data_group["trigger_telescope_list_list"][:]
+                except KeyError as exc:
+                    raise KeyError(
+                        "One or more required datasets are missing from the 'data' group."
+                    ) from exc
+            else:
+                raise KeyError("data group is missing from the HDF5 file.")
+
+    def _generate_2d_histogram(self, x_data, y_data, x_bins, y_bins):
         """
-        self.angle_to_observing_position__triggered_showers_ = None
-        self.event_weight__ra3d__log10_e__ = None
+        Generate a 2D histogram from the given data.
 
-        for table in event_data_file_tables:
-            if (
-                "Title" in table.meta
-                and table.meta["Title"] == "angle_to_observing_position__triggered_showers_"
-            ):
-                self.angle_to_observing_position__triggered_showers_ = table
-            elif "Title" in table.meta and table.meta["Title"] == "event_weight__ra3d__log10_e__":
-                self.event_weight__ra3d__log10_e__ = table
+        Parameters
+        ----------
+        x_data : np.ndarray
+            Array of x-axis data.
+        y_data : np.ndarray
+            Array of y-axis data.
+        x_bins : np.ndarray
+            Array of bin edges for the x-axis.
+        y_bins : np.ndarray
+            Array of bin edges for the y-axis.
+
+        Returns
+        -------
+        np.ndarray
+            2D histogram array.
+        """
+        hist, _, _ = np.histogram2d(x_data, y_data, bins=[x_bins, y_bins])
+        return hist
 
-    def _compute_limits(
-        self, event_weight_array, bin_edges, loss_fraction, axis=0, limit_type="lower"
-    ):
+    def _compute_limits(self, hist, bin_edges, loss_fraction, axis=0, limit_type="lower"):
         """
         Compute the limits based on the loss fraction.
 
         Parameters
         ----------
-        event_weight_array : np.ndarray
-            Array of event weights.
+        hist : np.ndarray
```

Comment:
Does it actually need to be a 2D histogram? If I understand the code correctly in `compute_viewcone`, then a 1D histogram is filled there?